tuf/api/metadata.py

```diff
@@ -415,53 +415,102 @@ def remove_key(self, role: str, keyid: str) -> None:
             del self.keys[keyid]
 
 
+class MetaFile:
+    """A container with information about a particilur file.
+    Instances of Metafile are used as values in a dictionary called "meta" in
+    Timestamp and Snapshot:
+
+        meta: A dictionary that contains information about metadata files::
+            {
+                'meta_file_path_1': <MetaFile INSTANCE_1>.
+                'meta_file_path_2': <MetaFile INSTANCE_2>
+                ...
+            }
+    Attributes:
+        version: An integer indicating the version of the metafile.
+        length: An optional integer indicating the length of the metafile.
+        hashes: An optional dictionary containing hash algorithms and the
+            hashes resulted from applying them over the metafile::
+                'hashes': {
+                    '<HASH ALGO 1>': '<META FILE HASH 1>',
+                    '<HASH ALGO 2>': '<META FILE HASH 2>',
+                    ...
+                }
+
+    """
+
+    def __init__(self, version: int, length: Optional[int] = None,
+            hashes: Optional[JsonDict] = None) -> None:
+        self.version = version
+        self.length = length
+        self.hashes = hashes
+
+
+    def __eq__(self, other: Any) -> bool:
+        """Used to compare objects by their values instead of by their
+        addresses."""
+        if not isinstance(other, MetaFile):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
```

Comment:
Why not use type hints and (in a follow-up PR) validate that instead of doing an `isinstance` on the passed arg?

```suggestion
    def __eq__(self, other: 'MetaFile') -> bool:
        """Used to compare objects by their values instead of by their
        addresses."""
```

Reply:
I was not sure how to do it, because I couldn't use `MetaFile` directly, because it was not yet defined.
I will use your approach. :)

Reply:
See https://www.python.org/dev/peps/pep-0484/#forward-references for more info about that scenario.